### PR TITLE
Onboarding Project: Refactor — Utils

### PR DESCRIPTION
**Pull request by Jropp**
## What It Does

Moves phone settings (number length range, symbols, etc) into TITLE_CASE settings object.

The various settings for phone numbers were spattered across the code. This change makes it easier to change settings in one spot, thus making the code more maintainable.

## How To Test

Provide instructions on how to test this change. If there are tricky or edge cases that should be tested, mention those too.

Screenshots are great for highlighting visual changes.

## Notes/Caveats

Are there remaining issues or things this PR hasn't solved yet? Are there long-term implications?

## Checklist

Under penalty of public shaming, I avow that I:

- [x] Reviewed the diff to look for typos, whitespace errors, and console.log() statements
- [x] Verified that there are no compiler or linter errors or warnings
- [ ] Verified the build in production(optimized) mode
- [ ] Updated the docs, if necessary
- [x] Included the appropriate labels
- [ ] Referenced applicable issues (i.e. Closes #XXXX, Fixes #XXXX)
- [x] ARE YOU MERGING INTO THE CORRECT BRANCH!?

Browsers tested:

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] Internet Explorer

## Dependencies

- [ ] #195

### CODE REVIEW

File: src/api/utils.js [hunk 1]
```diff
@@ -1,13 +1,25 @@
+const phoneSettings = {
```
Comment (collincahill) [inline src/api/utils.js]: I think you may want to put this into a static getter within the class.

File: src/api/utils.js [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+  INTERNATIONAL_MAX_LENGTH: 13,
+  INTERNATIONAL_LENGTH: (length) => {return length > this.DOMESTIC_LENGTH},
+  DOMESTIC_SYMBOLS: [
+    { name: 'dash', loc: 4, text: '-' },
+    { name: 'spaceLoc', loc: 8, text: ' ' },
+    { name: 'leftParLoc', loc: 9, text: ')' },
+    { name: 'rightParLoc', loc: 13, text: '(' }
+  ],
+  INTERNATIONAL_SYMBOLS: [
+    { name: 'intPre', loc: reversedPhoneArray.length, text: '+' },
```
Comment (collincahill) [inline src/api/utils.js]: Where is `reversedPhoneArray` coming from in this scope?

File: src/api/utils.js [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
   static formatPhoneNumber(phoneNumber) {
-    const domesticPhoneLength = 10;
-    const isInternationalNumber = phoneNumber.length > domesticPhoneLength;
     const reversedPhoneArray = phoneNumber.split('').reverse();
-
     const domFormatted = this.insertDomesticSymbols(reversedPhoneArray);
     let displayFormat;
 
-    if (isInternationalNumber) {
+    if (this.INTERNATIONAL_LENGTH(phoneNumber.length)) {
```
Comment (collincahill) [inline src/api/utils.js]: If `phoneSettings` is a static getter you can do `this.phoneSettings.INTERNATIONAL_LENGTH`, otherwise, you'll just need to do `phoneSettings.INTERNATIONAL_LENGTH` here.

File: src/api/utils.js [hunk 4]
```diff
@@ -1,13 +1,25 @@
...
+    { name: 'spaceLoc', loc: 8, text: ' ' },
+    { name: 'leftParLoc', loc: 9, text: ')' },
+    { name: 'rightParLoc', loc: 13, text: '(' }
+  ],
+  INTERNATIONAL_SYMBOLS: [
+    { name: 'intPre', loc: reversedPhoneArray.length, text: '+' },
+    { name: 'intSpace', loc: 14, text: ' ' }
+  ]
+}
 export class ValidateUtil {
```
Comment (collincahill) [inline src/api/utils.js]: Looks like Webpack will want this to be `export default class ValidateUtil`

Comment (collincahill) [reply src/api/utils.js]: And you will need to import the class in `user-component.js` where it's being used

**Comment by collincahill** [comment]
@Jropp 
If you can get Webpack to bundle everything properly without the default statement, it's fine. It just gave me the error 🤔 I don't know all the inner workings and maybe I should read up on it!!

You should be able to do a static function. Either ought to work.
